module/files/helper.js

```diff
@@ -335,4 +338,177 @@ module.exports = class FilesHelper {
+          if (!Array.isArray(fileNames) || fileNames.length < 1) {
+            throw new Error('File names not given.');
+          }
+
+          let noOfMinutes = messageConstants.common.NO_OF_MINUTES;
+          let linkExpireTime = messageConstants.common.NO_OF_EXPIRY_TIME * noOfMinutes;
+          // Create an array of promises for signed URLs
+          // {sample response} : https://sunbirdstagingpublic.blob.core.windows.net/samiksha/reports/sample.pdf?sv=2020-10-02&st=2023-08-03T07%3A53%3A53Z&se=2023-08-03T08%3A53%3A53Z&sr=b&sp=w&sig=eZOHrBBH%2F55E93Sxq%2BHSrniCEmKrKc7LYnfNwz6BvWE%3D
+          const signedUrlsPromises = fileNames.map(async (fileName) => {
+            let file = folderPath && folderPath !== '' ? folderPath + fileName : fileName;
+            let response = {
+              file: file,
+              payload: { sourcePath: file },
+              cloudStorage: cloudStorageProvider.toUpperCase(),
+            };
+            response.downloadableUrl = await cloudClient.getDownloadableUrl(
+              bucketName,
+              file,
+              linkExpireTime, // Link ExpireIn
+            );
+            if (!serviceUpload) {
+              response.url = await cloudClient.getSignedUrl(
+                bucketName, // bucket name
+                file, // file path
+                linkExpireTime, // expire
+                actionPermission, // read/write
+              );
+            } else {
+              response.url = `${process.env.PUBLIC_BASE_URL}/${messageConstants.common.UPLOAD_FILE}?file=${file}`;
+            }
+
+            if (addDruidFileUrlForIngestion) {
+              // {sample response} : { type: 's3', uris: [ 's3://dev-mentoring/reports/cspSample.pdf' ] }
+              let druidIngestionConfig = await cloudClient.getFileUrlForIngestion(
+                bucketName, // bucket name
+                file, // file path
+              );
+              response.inputSource = druidIngestionConfig;
+            }
+            return response;
+          });
+
+          // Wait for all signed URLs promises to resolve
+          signedUrls = await Promise.all(signedUrlsPromises);
+          result[payloadIds[0]]['files'] = signedUrls;
+
+          return resolve({
+            message: messageConstants.apiResponses.URL_GENERATED,
+            data: result,
+          });
+        } catch (error) {
+          return resolve({
+            status: httpStatusCode['bad_request'].status,
+            message: messageConstants.apiResponses.FAILED_PRE_SIGNED_URL,
+            result: {},
+          });
+        }
+      } catch (error) {
+        return reject(error);
+      }
+    });
+  }
+  /**
+   * Get Downloadable URL from cloud.
+   * @method
+   * @name getDownloadableUrl
+   * @param {Array} payloadData       - payload for files data.
+   * @returns {JSON}                  - Response with status and message.
+   */
+
+  static getDownloadableUrl(payloadData) {
```

Comment:
@borkarsaish65 we can have two type buckets, private or public. if bucket is private this functions response won't work. that time we have to call getSignedUrl function with read access. you can verify tis by testing this with private bucket credentials . 
And also the bucket type (public/private)should be configurable via env. This also you can check here : https://github.com/ELEVATE-Project/project-service/blob/develop/controllers/v1/files.js

Reply:
Done. Here when using the private configuration. the response structure was not similar to public bucket condition.
I have made it similar so now the response is consitent for both type of bucket.